nemo/collections/nlp/models/language_modeling/megatron/falcon/falcon_decoder_layer.py

```diff
@@ -0,0 +1,284 @@
+        self.config: TransformerConfig = config
+
+        self.layer_number = layer_number + self._get_layer_offset()
+
+        self.self_attn_mask_type = self_attn_mask_type
+
+        if hasattr(self.config, 'new_decoder_architecture'):
+            self.new_decoder_architecture = self.config.new_decoder_architecture
+        else:
+            self.new_decoder_architecture = None
+
+        if hasattr(self.config, 'parallel_attention'):
+            self.parallel_attention = self.config.parallel_attention
+        else:
+            self.parallel_attention = None
+
+        ## [Module 1: Input Layernorm] Optional Layernorm on the input data
+        # TODO: add pytorch only layernorm
+        self.input_layernorm = build_module(
+            submodules.input_layernorm,
+            config=self.config,
+            hidden_size=self.config.hidden_size,
+            eps=self.config.layernorm_epsilon,
+            persist_layer_norm=self.config.persist_layer_norm,
+            sequence_parallel=self.config.sequence_parallel,
+            zero_centered_gamma=self.config.layernorm_zero_centered_gamma,
+            normalization=self.config.normalization,
+        )
+
+        ## [Module 2: SelfAttention]
+        self.self_attention = build_module(submodules.self_attention, config=self.config, layer_number=layer_number,)
+
+        ## [Module 3: BiasDropoutFusion] Optional
+        self.self_attn_bda = build_module(submodules.self_attn_bda)
+
+        ## [Module 4: Post SelfAttention] Optional Layernorm after self-attn
+        if self.new_decoder_architecture or self.parallel_attention:
+            self.post_self_attn_layernorm = None
+        else:
+            self.post_self_attn_layernorm = build_module(
+                submodules.post_self_attn_layernorm,
+                config=self.config,
+                hidden_size=self.config.hidden_size,
+                eps=self.config.layernorm_epsilon,
+                persist_layer_norm=self.config.persist_layer_norm,
+                sequence_parallel=self.config.sequence_parallel,
+                zero_centered_gamma=self.config.layernorm_zero_centered_gamma,
+                normalization=self.config.normalization,
+            )
+
+        ## [Module 5: pre mlp layernorm] Optional Layernorm before MLP, used in Falcon's new decoder architecture
+        self.pre_mlp_layernorm = (
+            build_module(
+                submodules.pre_mlp_layernorm,
+                config=self.config,
+                hidden_size=self.config.hidden_size,
+                eps=self.config.layernorm_epsilon,
+                persist_layer_norm=self.config.persist_layer_norm,
+                sequence_parallel=self.config.sequence_parallel,
+                zero_centered_gamma=self.config.layernorm_zero_centered_gamma,
+                normalization=self.config.normalization,
+            )
+            if self.new_decoder_architecture
+            else None
```

Comment:
Above it's done like:
```
if self.new_decoder_architecture or self.parallel_attention:
    self.post_self_attn_layernorm = None
else:
    ....
```
It would be good to use one style consistently.

Reply:
thanks. changed this block to if ... else ... style in the latest commit.